src/pip/_internal/vcs/git.py

```diff
@@ -259,12 +259,21 @@ def fetch_new(self, dest, url, rev_options):
         # type: (str, HiddenText, RevOptions) -> None
         rev_display = rev_options.to_display()
         logger.info('Cloning %s%s to %s', url, rev_display, display_path(dest))
-        self.run_command(make_command('clone', '-q', url, dest))
+        if self.get_git_version() >= (2, 17):
+            # Git added support for partial clone in 2.17
+            # https://git-scm.com/docs/partial-clone
+            # Speeds up cloning by functioning without a complete copy of repository
+            self.run_command(make_command(
+                'clone', '--filter=blob:none', '-q', url, dest,
+            ))
+        else:
+            self.run_command(make_command('clone', '-q', url, dest))
 
         if rev_options.rev:
             # Then a specific revision was requested.
             rev_options = self.resolve_revision(dest, url, rev_options)
             branch_name = getattr(rev_options, 'branch_name', None)
+            logger.info('Rev options %s, branch_name %s', rev_options, branch_name)
```

Comment:
Should this be shown by default? I feel it should be `debug` instead.

Reply:
debug makes sense - good point